transports/tcp/src/lib.rs

```diff
@@ -108,7 +110,24 @@ impl Transport for $tcp_config {
         async fn do_listen(cfg: $tcp_config, socket_addr: SocketAddr)
             -> Result<impl Stream<Item = Result<ListenerEvent<Ready<Result<$tcp_trans_stream, io::Error>>, io::Error>, io::Error>>, io::Error>
         {
-            let listener = <$tcp_listener>::bind(&socket_addr).await?;
+            let socket = if socket_addr.is_ipv4() {
+                Socket::new(Domain::ipv4(), Type::stream(), Some(socket2::Protocol::tcp()))?
+            } else {
+                let s = Socket::new(Domain::ipv6(), Type::stream(), Some(socket2::Protocol::tcp()))?;
+                s.set_only_v6(true)?;
+                s
+            };
+            if cfg!(target_family = "unix") {
+                socket.set_reuse_address(true)?;
+            }
+            socket.bind(&socket_addr.into())?;
+            socket.listen(1024)?; // we may want to make this configurable
```

Comment:
The default in Rust seems to be 128
https://github.com/rust-lang/rust/blob/c06e4aca19046b07d952b16e9f002bfab38fde6b/src/libstd/sys/unix/ext/net.rs#L823

```suggestion
            socket.listen(128)?; // we may want to make this configurable
```

Reply:
I was following [MIOs](https://crates.io/crates/mio) lead here as it is what tokio and async-std use.